giraffe/src/utils/fromFlux.ts

```diff
@@ -219,6 +375,51 @@ const splitChunks = (fluxCSV: string): string[] => {
   return chunks
 }
 
+/*
+  A Flux CSV response can contain multiple CSV files each joined by a newline.
+  This function splits up a CSV response into these individual CSV files.
+  See https://github.com/influxdata/flux/blob/master/docs/SPEC.md#multiple-tables.
+*/
+const splitChunksFast = (fluxCSV: string): string[] => {
+  const trimmedResponse = fluxCSV.trim()
+
+  if (trimmedResponse === '') {
+    return []
+  }
+
+  // Split the response into separate chunks whenever we encounter:
+  //
+  // 1. A newline
+  // 2. Followed by any amount of whitespace
+  // 3. Followed by a newline
+  // 4. Followed by a `#` character
+  //
+  // The last condition is [necessary][0] for handling CSV responses with
+  // values containing newlines.
+  //
+  // [0]: https://github.com/influxdata/influxdb/issues/15017
+  let curr = 0
+
+  let chunks = []
+  while (curr !== -1) {
+    const oldVal = curr
+    const nextIndex = trimmedResponse
+      .substring(curr, trimmedResponse.length)
+      .search(/\n\s*\n#/)
+    if (nextIndex === -1) {
+      chunks.push([oldVal, trimmedResponse.length])
+      curr = -1
+      break
+    } else {
+      chunks.push([oldVal, oldVal + nextIndex])
+      curr = oldVal + nextIndex + 3
+    }
+  }
+  chunks = chunks.map(([start, end]) => trimmedResponse.substring(start, end))
```

Comment:
this operation uses the same `regex` pattern to identify the position of the chunks and saves a position reference to the chunk beginning and end, using those to then create the array of chunks based on the original CSV, rather than the more memory-intensive `split().map()` approach in `splitChunks`